RecoTracker/MkFitCore/interface/HitStructures.h

```diff
@@ -0,0 +1,874 @@
+      lastHitIdx_ = -1;
+      nFoundHits_ = 0;
+    }
+
+    CombCandidate* combCandidate() const { return m_comb_candidate; }
+    void setCombCandidate(CombCandidate* cc) { m_comb_candidate = cc; }
+
+    int lastCcIndex() const { return lastHitIdx_; }
+    int nFoundHits() const { return nFoundHits_; }
+    int nMissingHits() const { return nMissingHits_; }
+    int nOverlapHits() const { return nOverlapHits_; }
+    int nTotalHits() const { return nFoundHits_ + nMissingHits_; }
+
+    void setLastCcIndex(int i) { lastHitIdx_ = i; }
+    void setNFoundHits(int n) { nFoundHits_ = n; }
+    void setNMissingHits(int n) { nMissingHits_ = n; }
+    void setNOverlapHits(int n) { nOverlapHits_ = n; }
+
+    int nInsideMinusOneHits() const { return nInsideMinusOneHits_; }
+    int nTailMinusOneHits() const { return nTailMinusOneHits_; }
+
+    void setNInsideMinusOneHits(int n) { nInsideMinusOneHits_ = n; }
+    void setNTailMinusOneHits(int n) { nTailMinusOneHits_ = n; }
+
+    int originIndex() const { return m_origin_index; }
+    void setOriginIndex(int oi) { m_origin_index = oi; }
+
+    void resetOverlaps() { m_overlap_hits.reset(); }
+    void considerHitForOverlap(int hit_idx, int module_id, float chi2) {
+      m_overlap_hits.consider_hit_for_overlap(hit_idx, module_id, chi2);
+    }
+    HitMatch* findOverlap(int hit_idx, int module_id) { return m_overlap_hits.find_overlap(hit_idx, module_id); }
+
+    // Inlines after definition of CombCandidate
+
+    HitOnTrack getLastHitOnTrack() const;
+    int getLastHitIdx() const;
+    int getLastHitLyr() const;
+
+    // For additional filter
+    int getLastFoundPixelHitLyr() const;
+    int getLastFoundHitLyr() const;
+    int nUniqueLayers() const;
+
+    int nLayersByTypeEncoded(const TrackerInfo& trk_inf) const;
+    int nHitsByTypeEncoded(const TrackerInfo& trk_inf) const;
+
+    int nPixelDecoded(const int& encoded) const { return encoded % 100; }
+    int nStereoDecoded(const int& encoded) const { return (encoded / 100) % 100; }
+    int nMonoDecoded(const int& encoded) const { return (encoded / 10000) % 100; }
+    int nMatchedDecoded(const int& encoded) const { return encoded / 1000000; }
+    int nTotMatchDecoded(const int& encoded) const {
+      return encoded % 100 + (encoded / 100) % 100 + (encoded / 10000) % 100 - encoded / 1000000;
+    }
+
+    void addHitIdx(int hitIdx, int hitLyr, float chi2);
+
+    HoTNode& refLastHoTNode();              // for filling up overlap info
+    const HoTNode& refLastHoTNode() const;  // for dump traversal
+
+    void incOverlapCount() { ++nOverlapHits_; }
+
+    Track exportTrack(bool remove_missing_hits = false) const;
+
+    void resetShortTrack() {
+      score_ = getScoreWorstPossible();
+      m_comb_candidate = nullptr;
+    }
+
+  protected:
+    CombCandidate* m_comb_candidate = nullptr;
+    HitMatchPair m_overlap_hits;
+
+    // using from TrackBase:
+    // short int lastHitIdx_
+    // short int nFoundHits_
```

Comment:
remove?

Reply:
removed what is not needed, clarified usage of base-class member lastHitIdx_

Addressed in Round 2 follow-up PR trackreco/cmssw#76 commit: https://github.com/trackreco/cmssw/pull/76/commits/d2c6a5ba4edbe357152710835beb41fb16faf51e